drivers/wdm/audio/hdaudbus_new/regfuncs.h

```diff
@@ -0,0 +1,87 @@
+static inline UINT8 read8(PVOID addr) {
+	UINT8 ret = *(UINT8*)addr;
```

Comment:
Please use `READ_REGISTER_` and 'WRITE_REGISTER_' routines when accessing MMIO memory. These functions are portable and have a memory barrier after each access on some architectures. `volatile` isn't always sufficient to ensure the correct ordering, but that depends on the hardware.